daemon/daemon/endpoint.go

```diff
@@ -125,73 +129,82 @@ func (d *Daemon) setEndpointSecLabel(endpointID *uint16, dockerID, dockerEPID st
-		return ep.DeepCopy(), nil
+	switch prefix {
+	case endpoint.CiliumIdPrefix:
+		numid, err := strconv.ParseInt(id, 10, 16)
+		if err != nil {
+			return nil, apierror.Error(GetEndpointIDInvalidCode, err)
+		}
+		return d.lookupCiliumEndpoint(uint16(numid)), nil
+
+	case endpoint.DockerUUIDPrefix:
+		return d.lookupDockerID(id), nil
+
+	case endpoint.DockerEndpointPrefix:
+		return d.lookupDockerEndpoint(id), nil
+
+	default:
+		return nil, apierror.New(GetEndpointIDInvalidCode, "Unknown endpoint prefix %s", prefix)
 	}
-	return nil, nil
 }
 
-// EndpointGetByDockerEPID returns a copy of the endpoint for the given dockerEPID, or nil
-// if the endpoint was not found.
-func (d *Daemon) EndpointGetByDockerEPID(dockerEPID string) (*endpoint.Endpoint, error) {
+func (d *Daemon) EndpointExists(id string) bool {
 	d.endpointsMU.RLock()
 	defer d.endpointsMU.RUnlock()
 
-	if ep := d.lookupDockerEndpoint(dockerEPID); ep != nil {
-		return ep.DeepCopy(), nil
-	}
-	return nil, nil
+	ep, err := d.lookupEndpointLocked(id)
+	return err == nil && ep != nil
 }
 
-// EndpointGet returns a copy of the endpoint for the given endpointID, or nil if the
-// endpoint was not found.
-func (d *Daemon) EndpointGet(endpointID uint16) (*endpoint.Endpoint, error) {
-	d.endpointsMU.RLock()
-	defer d.endpointsMU.RUnlock()
+type getEndpointList struct {
+	d *Daemon
+}
+
+func NewGetEndpointListHandler(d *Daemon) GetEndpointListHandler {
+	return &getEndpointList{d: d}
+}
+
+func (h *getEndpointList) Handle(params GetEndpointListParams) middleware.Responder {
+	h.d.endpointsMU.RLock()
+	defer h.d.endpointsMU.RUnlock()
 
-	if ep := d.lookupCiliumEndpoint(endpointID); ep != nil {
-		return ep.DeepCopy(), nil
+	eps := []*models.Endpoint{}
+	for k := range h.d.endpoints {
+		eps = append(eps, h.d.endpoints[k].GetModel())
 	}
 
-	return nil, nil
+	return NewGetEndpointListOK().WithPayload(eps)
 }
 
-// EndpointsGet returns a copy of all the endpoints or nil if there are no endpoints.
-func (d *Daemon) EndpointsGet() ([]endpoint.Endpoint, error) {
-	d.endpointsMU.RLock()
-	defer d.endpointsMU.RUnlock()
+type getEndpoint struct {
+	d *Daemon
+}
 
-	eps := []endpoint.Endpoint{}
-	epsSet := map[*endpoint.Endpoint]bool{}
-	for _, v := range d.endpoints {
-		epsSet[v] = true
-	}
-	if len(epsSet) == 0 {
-		return nil, nil
-	}
-	for k := range epsSet {
-		epCopy := k.DeepCopy()
-		eps = append(eps, *epCopy)
-	}
-	return eps, nil
+func NewGetEndpointIDHandler(d *Daemon) GetEndpointIDHandler {
+	return &getEndpoint{d: d}
 }
 
-func (d *Daemon) deleteEndpoint(endpointID uint16) {
+func (h *getEndpoint) Handle(params GetEndpointIDParams) middleware.Responder {
+	h.d.endpointsMU.RLock()
+	defer h.d.endpointsMU.RUnlock()
 
-	if ep := d.lookupCiliumEndpoint(endpointID); ep != nil {
-		delete(d.endpointsDocker, ep.DockerID)
-		delete(d.endpointsDockerEP, ep.DockerEndpointID)
-		delete(d.endpoints, endpointID)
+	if ep, err := h.d.lookupEndpointLocked(params.ID); err != nil {
+		return err
```

Comment:
Yes, see `lookupEndpointLocked()` implementation. The common error codes all match from all users of the function.